Update WordPress JS packages and react [MAILPOET-3654][MAILPOET-3532]

costasovo opened the pull request:
This PR covers both Jira tickets since @wordpress/element bumped react version.

Apart from buch of small fixes there are 4 notable changes:

1. Workaround for issue with different versions of @emotion sub deps between storybook and @wordpress/components. You can find more details in [the related commit](https://github.com/mailpoet/mailpoet/commit/d2d61038f52c979f2d7879e44d03b6357bc59ae9) message.
2. I merged webpack configs for settings, form_editor and admin_vendor to admin_vendo in webpack.config.js. After this change webpack still produces separate files for form_editor and settings, but it is now aware of shared packages. This also helps reducing plugin's zip package size. See [the related commit description](https://github.com/mailpoet/mailpoet/commit/dccbadc232c1cd1510d0b1f253aa729f6e3bf2c1) for some details.
3. I fixed width settings for column blocks. I found that this was actually broken also in currently released version
4. I added a support for padding settings in column block.

**Note for QA:**
Please check overall functionality of the editor and changes mentioned in points 3. and 4.

[MAILPOET-3654][MAILPOET-3532]

[MAILPOET-3654]: https://mailpoet.atlassian.net/browse/MAILPOET-3654

pavel-mailpoet commented on package.json:
```diff
@@ -84,10 +84,10 @@
     "parsleyjs": "^2.9.2",
     "prop-types": "^15.7.2",
-    "react": "^16.13.1",
+    "react": "^17.0.2",
```
🎉

pavel-mailpoet commented on assets/js/src/form_editor/components/editor.jsx:
```diff
@@ -74,6 +74,7 @@ export default () => {
   const editorSettings = {
     mediaUpload: canUserUpload ? uploadMedia : null,
     maxWidth: 580,
+    enableCustomSpacing: true,
```
🎉

pavel-mailpoet reviewed (approved):
Great job 👍

veljkho reviewed (changes requested):
Some issues found:
1. No custom color link for background color of the Subscribe button
2. No custom color link for background color of the Form itself
3. No custom color link for background color of the Heading block
...but that's actually one issue for all.

costasovo commented:
@pavel-mailpoet During the rebase I ended up updating @wordpress/* packages again. I added 2 more commits. The first one addresses changes in block-editor settings and also addresses the issue found by @veljkho 

@veljkho As I updated the packages to even a newer version, could you please do a complete check of the PR again?

veljkho reviewed (changes requested):
Some issues found:
- Can't see applied background color to column single block
- Can't see applied font color to column single block
- Can't see all the blocks inside [Browse All blocks sidebar](https://d.pr/i/Ae2Y9n)
info: this is happening when you click + button inside a column block and then Browse All.
